src/Compilers/CSharp/Portable/Symbols/Source/TypeParameterConstraintClause.cs

```diff
@@ -197,65 +183,74 @@ static bool isValueType(TypeParameterSymbol thisTypeParameter, ArrayBuilder<Type
                     }
                 }
 
-                isValueTypeOverrideBuilder.Add(thisTypeParameter, result);
+                isValueTypeMap.Add(thisTypeParameter, result);
                 return result;
             }
-
-            static void adjustConstraintTypes(Symbol container, ImmutableArray<TypeWithAnnotations> constraintTypes, IReadOnlyDictionary<TypeParameterSymbol, bool> isValueTypeOverride)
-            {
-                foreach (var constraintType in constraintTypes)
-                {
-                    constraintType.VisitType(null, (type, args, unused2) =>
-                    {
-                        if (type.DefaultType is TypeParameterSymbol typeParameterSymbol && typeParameterSymbol.ContainingSymbol == (object)args.container)
-                        {
-                            type.TryForceResolve(args.isValueTypeOverride[typeParameterSymbol]);
-                        }
-                        return false;
-                    }, typePredicate: null, arg: (container, isValueTypeOverride), canDigThroughNullable: false, useDefaultType: true);
-                }
-            }
         }
-    }
 
-    internal static class TypeParameterConstraintClauseExtensions
-    {
-        internal static bool HasValue(this ImmutableArray<TypeParameterConstraintClause> constraintClauses, bool canIgnoreNullableContext)
+        internal static SmallDictionary<TypeParameterSymbol, bool> BuildIsReferenceTypeFromConstraintTypesMap(Symbol container, ImmutableArray<TypeParameterSymbol> typeParameters,
```

Comment:
>FromConstraintTypesMap [](start = 87, length = 22)

Is there a reason this method name contains "FromConstraintTypes" and the previous method name does not?

Reply:
> Is there a reason this method name contains "FromConstraintTypes" and the previous method name does not?

Yes, this method only takes constraint types into account and the previous method does not.

---
In reply to: [521604502](https://github.com/dotnet/roslyn/pull/49261#discussion_r521604502) [](ancestors = 521604502)